scripts/DGSPlanner/MatrixUBInputWidget.py

```diff
@@ -110,6 +115,34 @@ def loadIsawUBDialog(self):
         except:
             mantid.logger.error("Could not open the file, or not a valid UB matrix")
 
+    def loadNexusUBDialog(self):
+        # pylint: disable=bare-except
+        try:
+            fname = QtGui.QFileDialog.getOpenFileName(\
+                    self, 'Open Nexus file to extract UB matrix',\
+                    filter=QString('Nexus file (*.nxs.h5);;All Files (*)'))
+            this_here_UB = LoadNexusUB(str(fname))
+            UB00 = self.UBmodel.createIndex(0,0)
```

Comment:
Just create an oriented lattice object, set the UB on it, and update the UBmodel with the oriented lattice.
See https://github.com/mantidproject/mantid/blob/master/scripts/DGSPlanner/MatrixUBInputWidget.py#L106-L108